### DISCUSSION TITLE
Ability to poll a folder from a cloud storage solution

### DISCUSSION THREAD

**Discussion by WoosterInitiative** [jonaswinkler/paperless-ng#1358] [Feature Requests]
The scanner that I have does not have an FTP option, it instead has the ability to upload to cloud services (like Google Drive). This works OK, but does require an additional step for me to upload those files to paperless.

I envision having a folder in Google Drive (or similar) that paperless monitors and does a full import of that file (not using Google Drive for storage, just as watched "consume" folder).

I'm going to attempt to add the ability in my own fork, and if I figure it out and it works well, I may do a pull request. This project is definitely a stretch for me, though.

**Comment by TobenderZephyr** [comment]
Maybe this is not exactly what you want, but as a dirty workaround you could mount Google Drive as a virtual filesystem and point your consume folder into it?

See this link on how to do that (for terminals I would prefer no 3): https://ostechnix.com/how-to-mount-google-drive-locally-as-virtual-file-system-in-linux/

EDIT: I just realized there was already a Feature Request which fits into the same category: S3 backend for files #883

**Reply by WoosterInitiative** [reply]
Thanks for that, didn't see that request.

I've also looked at something that could just copy files into the local **consume** for me. That would not require any changes to NG and would do what I want.

If anyone knows of a service like that that monitors a GDrive/OneDrive/etc. folder and copies/moves new files (once) into a local directory, particularly in Linux Server (no GUI), I would appreciate it.

**Reply by TobenderZephyr** [reply]
What is preventing you from using rclone for this? This should fit perfectly to your requirement. See the first link i posted above.
